apps/src/p5lab/spritelab/CoreLibrary.js

```diff
@@ -100,8 +101,37 @@ export default class CoreLibrary {
     width = Math.max(width, 50);
     const height = lines.length * textSize + padding * 2;
 
+    let triangleSize = 10;
+    let tipX = x;
+
+    // For the calculations below, keep in mind that x and y are located at the horizontal center and the top of the sprite, respectively.
+    // In other words, x and y indicate the default position of the bubble's triangular tip.
+    if (y > 400) {
+      y = 400;
+    }
+    if (y - height - triangleSize < 1) {
+      triangleSize = Math.max(1, y - height);
+      y = height + triangleSize;
+    }
+    if (x - width / 2 < 1) {
+      tipX = Math.max(x, 8 + triangleSize);
```

Comment:
what does 8 signify here? it looks like we're using it on line 121 as well so it should likely be a constant (just scoped to this function, but giving it a name makes it clearer what we mean here)

Reply:
This accounts for the arced corners, based on `p5.rect(minX, minY, width, height, 8);`

Reply:
ahhhh, okay. we should add that to the configuration object if the caller has a dependency on it. so in CoreLibrary, we can use that value and pass it to speechBubble to ensure they stay in sync, but still have a default value of 8 for the corner radius within the speechBubble function itself

Reply:
Can do.